powerauth-push-client/src/main/java/io/getlime/push/client/PushServerClient.java

```diff
@@ -295,21 +295,24 @@ public boolean addUsersToCampaign(Long campaignId, List<String> users) throws Pu
 
     /**
      * Get a list of users in paged format from specific campaign
+     * Suppressed warning "unchecked". Unchecked warning assignment is actually not right.
+     * Jackson mapper, returned by @checkHttpStatus, can't recognize generic internals of TypeReference
      *
      * @param campaignId Identifier of campaign.
      * @param page Page number.
      * @param size Size of elements per page.
      * @return Page of users specified with params.
      */
-    public ObjectResponse<PagedResponse<ListOfUsersFromCampaignResponse>> getListOfUsersFromCampaign(Long campaignId, int page, int size) throws PushServerClientException {
+    @SuppressWarnings("unchecked")
+    public ListOfUsersPagedResponse<ListOfUsersFromCampaignResponse> getListOfUsersFromCampaign(Long campaignId, int page, int size) throws PushServerClientException {
         try {
             String campaignIdSanitized = URLEncoder.encode(String.valueOf(campaignId), "utf-8");
-            TypeReference<ObjectResponse<PagedResponse<ListOfUsersFromCampaignResponse>>> typeReference = new TypeReference<ObjectResponse<PagedResponse<ListOfUsersFromCampaignResponse>>>() {
+            TypeReference<ListOfUsersPagedResponse<ListOfUsersFromCampaignResponse>> typeReference = new TypeReference<ListOfUsersPagedResponse<ListOfUsersFromCampaignResponse>>() {
```

Comment:
Why? This is just confused code. We should have `PagedResponse<T>` that extends `ObjectResponse<List<T>>` and contains `List<T> items`.

Wrapping wrapped response object makes no sense...

Reply:
As explained above, mapping multiple generic objects wrapped in TypeReference can't go deeper than 2 times, so jacksonMapper won't recognize objects inside (e.g. ObjectResponse<List<T>> T won't be recognized). Anyway our ListOfUsersPagedResponse, extended with ObjectResponse ,is simple List of ListOfUsersCampaignResponse (naming might be confusing) . Had to create own class for using it's own deserializer. Because of to preserve PagedResponse template for possible further use I've made these changes.

Reply:
The code should look like:

```java
TypeReference<PagedResponse<ListOfUsersFromCampaignResponse>> typeReference = ...
```

Am I missing something?

Reply:
Yep, our client method working with ObjectResponse, either solve prolem as is, or completely rework client-server.

Reply:
OK, let's discuss on Friday. You are missing the point. The client works with whatever server publishes.